frontend/src/responseQuizView/responseQuizView.js

```diff
@@ -0,0 +1,343 @@
+import React from 'react';
+import './responseQuizView.scss';
+// import ReactTooltipDefaultExport from 'react-tooltip';
+import * as PropTypes from 'prop-types';
+import {
+    Button,
+    ToggleButton,
+    ToggleButtonGroup,
+} from 'react-bootstrap';
+
+import { Navbar, Footer, LoadingPage } from '../UILibrary/components';
+
+// This dictionary maps specific error types to their respective error message.
+const ErrorDescriptions = {
+    'backwards': 'The letters are reversed',
+    'verb-conjugation': 'One or more verbs are incorrectly conjugated',
+    'comma-splice': 'Two independent clauses are incorrectly joined by a comma',
+    'run-on': 'There is a run on sentence',
+    'homophone': 'Misuse of a similar sounding word (homophone)',
+};
+
+export class ResponseQuizView extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            data: null,
+            question: 1,
+            userAnswers: {},
+            score: 0,
+            graded: false,
+        };
+    }
+
+    async componentDidMount() {
+        try {
+            const response = await fetch(`/api/get_response_quiz_data/${this.props.textID}/`);
+            const data = await response.json();
+            this.setState({ data: data });
```

Comment:
This is a relatively minor point, but I'd recommend naming this something more descriptive than `data`. Just reading this `componentDidMount` offers no information about the shape of this data.

Reply:
Renamed to "questionData" for the sake of clarity.